api/src/test/java/org/openmrs/module/fhir/api/diagnosticreport/handler/RadiologyHandlerTest.java

```diff
@@ -0,0 +1,262 @@
+     * Creates and returns Encounter, given the patient's Uuid
+     * 
+     * @param patient_uuid Uuid of the patient 
+     * @return Encounter object
+     */
+    private Encounter createEncounter(String patient_uuid)
+    {
+        Encounter encounter = new Encounter();
+        encounter.setLocation(new Location(1));
+        encounter.setEncounterType(new EncounterType(1));
+        encounter.setEncounterDatetime(new Date());
+        encounter.setPatient(patientService.getPatientByUuid(patient_uuid));
+
+        EncounterRole role = new EncounterRole();
+        role.setName("role");
+        role = Context.getEncounterService().saveEncounterRole(role);
+
+        Provider provider = new Provider();
+        provider.setIdentifier("id1");
+        provider.setPerson(newPerson("name"));
+        provider = Context.getProviderService().saveProvider(provider);
+
+        Provider provider2 = new Provider();
+        provider2.setIdentifier("id2");
+        provider2.setPerson(newPerson("name2"));
+        provider2 = Context.getProviderService().saveProvider(provider2);
+
+        encounter.addProvider(role, provider);
+        encounter.addProvider(role, provider2);
+
+        EncounterType encounterType = new EncounterType("test", "test encounter type");
+        Context.getEncounterService().saveEncounterType(encounterType);
+        encounter.setEncounterType(encounterType);
+
+        Context.getAdministrationService().setGlobalProperty("fhir.encounter.encounterRoleUuid", role.getUuid());
+
+        EncounterService es = Context.getEncounterService();
+        es.saveEncounter(encounter);
+
+        encounterService.saveEncounter(encounter);
+        return encounter;
+    }
+    
+    /**
+     * @see RadiologyHandler#getServiceCategory
+     */
+    @Test
+    public void getServiceCategory_shouldReturnRAD()
+    {
+    	RadiologyHandler radiologyHandler=new RadiologyHandler();
+    	assertEquals(radiologyHandler.getServiceCategory(),"RAD");
+    }
+    
+    /**
+     * @see RadiologyHandler#getFHIRDiagnosticReportById
+     */
+    @Test
+    public void getFHIRDiagnosticReportById_shouldReturnNonNullDiagnosticReport() {
+        RadiologyHandler  radiologyHandler = new RadiologyHandler();
+        Patient patient = createPatient();
+        Encounter encounter = createEncounter(patient.getUuid());        
+        assertNotNull(radiologyHandler.getFHIRDiagnosticReportById(encounter.getUuid()));
+    }
+    
+    @Test
+    public void getFHIRDiagnosticReportBySubjectName_retrunListOfDiagnosticReports()
+    {
+    	RadiologyHandler radiologyHandler=new RadiologyHandler();
+    	Patient patient=createPatient();
+    	Encounter encounter = createEncounter(patient.getUuid());
+    	String name="";
+    	for(PersonName i:patient.getNames())
+    	{
+    	name=i.getGivenName();
+    	break;
+    	}
+    	try {
+       assertNotNull(radiologyHandler.getFHIRDiagnosticReportBySubjectName(name));
+    }
+    	catch(Exception e)
+    	{
```

Comment:
@haripriya999 , thank you very much for the great work. Quick question, why would you need to catch an `Exception` ?

Reply:
@Ruhanga  Thank you for reviewing my code :) . There seems to be some problem with the variable, serverBase in RadiologyHandler, beacause of which a null pointer exception was being throw. I understood the root cause of the problem but I'm unsure of what the variable value should be set to. Hope this Talk would provide further details- https://talk.openmrs.org/t/error-variable-serverbase-in-radiologyhandler-is-returning-null/22511 . Also, I think we can create a new issue for this.